wallet/test/Test/Pos/Wallet/Web/Util.hs

```diff
@@ -0,0 +1,210 @@
+-- Block utils
+----------------------------------------------------------------------------
+
+-- | Gen blocks in WalletProperty
+wpGenBlocks
+    :: (HasCompileInfo, HasConfigurations)
+    => Maybe BlockCount
+    -> EnableTxPayload
+    -> InplaceDB
+    -> WalletProperty (OldestFirst [] Blund)
+wpGenBlocks blkCnt enTxPayload inplaceDB = do
+    params <- genBlockGenParams blkCnt enTxPayload inplaceDB
+    g <- pick $ MkGen $ \qc _ -> qc
+    lift $ modifyStateLock HighPriority "wpGenBlocks" $ \prevTip -> do
+        blunds <- evalRandT (genBlocks params) g
+        case nonEmpty $ getOldestFirst blunds of
+            Just nonEmptyBlunds -> do
+                let tipBlockHeader = nonEmptyBlunds ^. _neLast . _1 . blockHeader
+                lastKnownHeader <- view (lensOf @LastKnownHeaderTag)
+                atomically $ writeTVar lastKnownHeader (Just tipBlockHeader)
+                progressHeader <- view (lensOf @ProgressHeaderTag)
+                atomically $ do
+                    void $ tryTakeTMVar progressHeader
+                    putTMVar progressHeader tipBlockHeader
+                pure (tipBlockHeader ^. headerHashG, blunds)
+            Nothing -> pure (prevTip, blunds)
+
+wpGenBlock
+    :: (HasCompileInfo, HasConfigurations)
+    => EnableTxPayload
+    -> InplaceDB
+    -> WalletProperty Blund
+wpGenBlock = fmap (unsafeHead . toList) ... wpGenBlocks (Just 1)
+
+----------------------------------------------------------------------------
+-- Wallet test helpers
+----------------------------------------------------------------------------
+
+-- | Import some nonempty set, but not bigger than 10 elements, of genesis secrets.
+-- Returns corresponding passphrases.
+importSomeWallets :: (HasConfigurations, HasCompileInfo) => WalletProperty [PassPhrase]
+importSomeWallets = do
+    let secrets =
+            map (view _2) .
+            gsSecretKeysPoor .
+            fromMaybe (error "Generated secrets are unknown") $ generatedSecrets
+    -- pva701: does sublistOf generate non-empty list?
+    (encSecrets, passphrases) <- pick $ do
+        seks <- take 10 <$> sublistOf secrets
+        let l = length seks
+        passwds <- vectorOf l arbitrary
+        pure (seks, passwds)
+    let wuses = map mkGenesisWalletUserSecret encSecrets
+    lift $ mapM_ (uncurry importWalletDo) (zip passphrases wuses)
+    skeys <- lift getSecretKeysPlain
+    assertProperty (not (null skeys)) "Empty set of imported keys"
+    pure passphrases
+
+-- | Take passphrases of our wallets
+-- and return some address from one of our wallets
+-- BE CAREFUL: this functions might take long time b/c it uses @deriveLvl2KeyPair@
+deriveRandomAddress :: [PassPhrase] -> WalletProperty (CId Addr)
+deriveRandomAddress passphrases = do
+    skeys <- lift getSecretKeysPlain
+    let l = length skeys
+    assert (l > 0)
+    walletIdx <- pick $ choose (0, l - 1)
+    let sk = skeys !! walletIdx
+    let psw = passphrases !! walletIdx
```

Comment:
Minor: carrying passphrases everywhere looks a bit cumbersome. Would be really nice to keep it in `WalletProperty` and extract some helper functions which require both `sk` and `pass`

Reply:
makes sense, but I wouldn't do it before it will needed